syntax = "proto3";
package ankiaddon.backend;

import "generic.proto";

service BackendService {
  rpc GetTatoebaLanguages(generic.Empty) returns (GetTatoebaLanguagesResponse);
  rpc DownloadTatoebaSentences(DownloadTatoebaSentencesRequest) returns (generic.Empty);
  rpc GetTatoebaDownloadProgress(generic.Empty) returns (TatoebaDownloadProgress);
}

message GetTatoebaLanguagesResponse {
  repeated Language languages = 1;
}

message Language {
  string code = 1;
  string name = 2;
}

message DownloadTatoebaSentencesRequest {
  string language = 1;
}

message TatoebaDownloadProgress {
  float progress = 1;
  string message = 2;
  bool is_error = 3;
  bool finished = 4;
}
